### PR TITLE
fix(ui): use timezone in downtime popin

### PR DESCRIPTION
**Pull request by kduret**
## Description

Take into account user timezone in downtime popin

/!\ this does not fix the table columns which do not use timezone

note : did not find a way to unit test timezone in popin

**Fixes** MON-5122

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

- Set a timezone to the user
- Add a downtime to a resource and check that dates and times take into account user timezone

### CODE REVIEW

File: www/front_src/src/Resources/Actions/Resource/Downtime/Dialog.tsx [hunk 1]
```diff
@@ -94,6 +97,14 @@ const DialogDowntime = ({
...
   };
 
+  React.useEffect(() => {
+    moment.locale(locale);
+  }, [locale]);
+
+  React.useEffect(() => {
+    moment.tz.setDefault(timezone);
+  }, [timezone]);
+
```
Comment (bdauria) [inline www/front_src/src/Resources/Actions/Resource/Downtime/Dialog.tsx:107]: I think these effects are a bit superfluous. I think we should load the dialog until all data needed are available (including locale and timezone).

Comment (kduret) [reply www/front_src/src/Resources/Actions/Resource/Downtime/Dialog.tsx:107]: I could put it in a single useEffect with "loading" as dependency, but I don't think it's the best way.
So I think I don't understand what is exactly your point ?

Comment (bdauria) [reply www/front_src/src/Resources/Actions/Resource/Downtime/Dialog.tsx:107]: No that's not what I mean. I mean that in the Dialog, locale and timezone are not supposed to be null, so we should have only one hook call setting them at the beginning. Also the Dialog is not supposed to be usable if they're not available yet

Comment (kduret) [reply www/front_src/src/Resources/Actions/Resource/Downtime/Dialog.tsx:107]: we'll probably refactor this once user information are stored in global context

**Pull request merged**